lib/breakers.rb

```diff
@@ -0,0 +1,28 @@
+require 'breakers/client'
+require 'breakers/outage'
+require 'breakers/service'
+require 'breakers/uptime_middleware'
+require 'breakers/version'
+
+require 'faraday'
+
+module Breakers
+  Faraday::Middleware.register_middleware(breakers: lambda { UptimeMiddleware })
+
+  # rubocop:disable Style/AccessorMethodName
+  def self.set_client(client)
+    @client = client
+  end
+  # rubocop:enable Style/AccessorMethodName
+
+  def self.client
+    @client
+  end
+
+  def self.new_connection(url: nil, adapter: Faraday.default_adapter)
+    Faraday.new(url: url_base) do |conn|
+      conn.use :breakers, @client
+      conn.adapter adapter
+    end
+  end
+end
```

Comment:
That code should have been removed and is gone now. I was trying to be too clever with creating Faraday connections automatically, but in reality I think we just want users of the gem to add the middleware into their existing Faraday connections.